api/policies/sessionAuth.js

```diff
@@ -1,8 +1,40 @@
-module.exports = function(req, res, next) {
-  req.session.authRedirectPath = undefined
+const moment = require('moment');
+const GitHub = require('../services/GitHub');
+
+// TODO: Move this to another file and export it for easier testing
+function isAuthDatePastThreshold(authDate, options = {}) {
+  const threshAmount = options.threshAmount || 30;
+  const threshUnits = options.threshUnits || 'minutes';
+  return moment().isAfter(moment(authDate).add(threshAmount, threshUnits));
+}
+
+// TODO: new tests
+module.exports = function sessionAuth(req, res, next) {
+  req.session.authRedirectPath = undefined;
   if (req.session.authenticated) {
+    const lastAuthenticatedAt = req.session.authenticatedAt;
+    if (!lastAuthenticatedAt || isAuthDatePastThreshold(lastAuthenticatedAt)) {
+      // check that the authenticated user is still a member of a valid GitHub org
+      return GitHub.validateUser(req.user.githubAccessToken)
+        .then(() => {
+          // reset authenticatedAt to the current time
+          req.session.authenticatedAt = new Date();
+          return next();
+        })
+        .catch(() => {
+          // log the user out and destroy their session
+          req.logout();
+          req.session.destroy();
+          return res.forbidden({
+            message: 'You are not permitted to perform this action.',
+          });
+        });
+    }
+    // else
     return next();
-  } else {
-    return res.forbidden('You are not permitted to perform this action. Are you sure you are logged in?');
   }
+  // else
+  return res.forbidden({
+    message: 'You are not permitted to perform this action. Are you sure you are logged in?',
```

Comment:
minor fix. the error handler looks for an object with `message` to pass along with the error response.